python/ray/util/client/worker.py

```diff
@@ -556,6 +581,11 @@ def terminate_actor(self, actor: ClientActorHandle,
         if not isinstance(actor, ClientActorHandle):
             raise ValueError("ray.kill() only supported for actors. "
                              "Got: {}.".format(type(actor)))
+        if actor.actor_ref.client_id != self._client_id:
+            raise ValueError(
```

Comment:
Can you tell the user the actual client it is from?

Reply:
Added `address` in the error message.